servicetalk-http-netty/src/main/java/io/servicetalk/http/netty/DefaultSingleAddressHttpClientBuilder.java

```diff
@@ -538,6 +555,13 @@ private static StreamingHttpConnectionFilterFactory appendConnectionFilter(
     @Override
     public DefaultSingleAddressHttpClientBuilder<U, R> retryServiceDiscoveryErrors(
             final ServiceDiscoveryRetryStrategy<R, ServiceDiscovererEvent<R>> retryStrategy) {
+        this.deprecatedServiceDiscovererRetryStrategy = requireNonNull(retryStrategy);
```

Comment:
Instead of opening a possibility for users to have two strategies at the same time, consider nullifying the other one when a builder method is invoked. Clarify that in javadoc. You can also log at warn level if you feel it's necessary.